File: SQLiteRepository/SQLiteQuery/GetQuery.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;

namespace SQLiteRepository.SQLiteQuery
{
    public class GetQuery : ISqliteQuery 
    {
        public string TableName { get; set; }
        public List<string> Columns { get; set; }
        public string Path { get; set; }
        public string Where { get; set; }
        public Tuple<string, List<string>> WhereIn { get; set; }
        public List<string> OrderBy { get; set; }
        public IQueryResult Execute()
        {
            return GetSqlite.GetData(this);
        }
        public IQueryResult Execute(SQLiteConnection conn, SQLiteTransaction tr)
        {
            return GetSqlite.GetData(conn, tr, this);
        }
    }
}

File: SQLiteAdapters/GetQuery.vb
Imports System.Data.SQLite


Public Class GetQuery
    Inherits BaseQuery
    Public Overrides Function Execute() As IQueryResult
        Return GetSqlite.GetData(Me)
    End Function
    Public Overrides Function Execute(conn As SQLiteConnection, tr As SQLiteTransaction) As IQueryResult
        Return GetSqlite.GetData(conn, tr, Me)
    End Function
     
End Class